lib/xandra/calendar.ex

```diff
@@ -0,0 +1,180 @@
+defmodule Xandra.Calendar do
+  if Version.match?(System.version(), ">= 1.6.0") do
+    def time_from_nanoseconds(nanoseconds) do
+      Time.add(~T[00:00:00], nanoseconds, :nanosecond)
+    end
+  else
+    def time_from_nanoseconds(nanoseconds) do
+      nanoseconds
+      |> DateTime.from_unix!(:nanoseconds)
+      |> DateTime.to_time()
+    end
+  end
```

Comment:
Instead of relying on the version it would be good to do `function_exported?(Time, :add, 3)` here and leave a TODO to remove it when we drop support for < 1.6.